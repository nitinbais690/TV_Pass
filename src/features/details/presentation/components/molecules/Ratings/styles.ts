import { scale } from 'qp-common-ui';
import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    ratings: {
        flexDirection: 'row',
        flexWrap: 'wrap',
        justifyContent: 'flex-start',
        alignItems: 'flex-start',
        marginTop: scale(16),
        marginBottom: scale(16),
        marginHorizontal: scale(16),
    },
});
