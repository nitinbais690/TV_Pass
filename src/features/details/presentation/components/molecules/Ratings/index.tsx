import React from 'react';
import { View } from 'react-native';
import Rating from '../../atoms/Rating';
import { styles } from './styles';

export default function Ratings(props: RatingsProps) {
    return (
        <View style={styles.ratings}>
            {props.ratings &&
                props.ratings.length > 0 &&
                props.ratings.map(rating => <Rating key={rating} rating={rating} />)}
        </View>
    );
}

export interface RatingsProps {
    ratings: string[];
}
