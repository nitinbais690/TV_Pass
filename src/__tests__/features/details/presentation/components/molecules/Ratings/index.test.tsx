import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import Ratings from 'features/details/presentation/components/molecules/Ratings';

jest.mock('core/presentation/hooks/use-app-colors');

describe('ratings', () => {
    const ratings: string[] = ['Premium', '18+', 'U/A'];

    test('renders correctly', () => {
        const { container } = render(<Ratings ratings={ratings} />);
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render empty list', () => {
        const { container } = render(<Ratings ratings={[]} />);
        expect(toJSON(container)).toMatchSnapshot();
    });
});
