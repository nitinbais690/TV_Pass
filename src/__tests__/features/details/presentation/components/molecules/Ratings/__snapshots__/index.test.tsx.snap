// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ratings render empty list 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
        "marginBottom": 23.314285714285713,
        "marginHorizontal": 23.314285714285713,
        "marginTop": 23.314285714285713,
      }
    }
  />
</View>
`;

exports[`ratings renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
        "marginBottom": 23.314285714285713,
        "marginHorizontal": 23.314285714285713,
        "marginTop": 23.314285714285713,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1A1D1E",
          "borderColor": "#35393F",
          "borderRadius": 7.285714285714286,
          "borderStyle": "solid",
          "borderWidth": 1.457142857142857,
          "elevation": 3.934285714285714,
          "flexDirection": "row",
          "height": 26.228571428571428,
          "justifyContent": "center",
          "marginEnd": "2%",
          "paddingHorizontal": 10.2,
          "shadowColor": "#40000000",
          "shadowOffset": Object {
            "height": 3.934285714285714,
            "width": 0,
          },
          "shadowRadius": 7.285714285714286,
        }
      }
    >
      <SvgMock
        style={
          Object {
            "marginEnd": 10.2,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 12,
            "fontWeight": "600",
            "letterSpacing": 0.13,
          }
        }
      >
        Premium
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1A1D1E",
          "borderColor": "#35393F",
          "borderRadius": 7.285714285714286,
          "borderStyle": "solid",
          "borderWidth": 1.457142857142857,
          "elevation": 3.934285714285714,
          "flexDirection": "row",
          "height": 26.228571428571428,
          "justifyContent": "center",
          "marginEnd": "2%",
          "paddingHorizontal": 10.2,
          "shadowColor": "#40000000",
          "shadowOffset": Object {
            "height": 3.934285714285714,
            "width": 0,
          },
          "shadowRadius": 7.285714285714286,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 12,
            "fontWeight": "600",
            "letterSpacing": 0.13,
          }
        }
      >
        18+
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1A1D1E",
          "borderColor": "#35393F",
          "borderRadius": 7.285714285714286,
          "borderStyle": "solid",
          "borderWidth": 1.457142857142857,
          "elevation": 3.934285714285714,
          "flexDirection": "row",
          "height": 26.228571428571428,
          "justifyContent": "center",
          "marginEnd": "2%",
          "paddingHorizontal": 10.2,
          "shadowColor": "#40000000",
          "shadowOffset": Object {
            "height": 3.934285714285714,
            "width": 0,
          },
          "shadowRadius": 7.285714285714286,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "ProximaNova-Regular",
            "fontSize": 12,
            "fontWeight": "600",
            "letterSpacing": 0.13,
          }
        }
      >
        U/A
      </Text>
    </View>
  </View>
</View>
`;
